PSKoans/Koans/IntroTraining/AboutOOP.koans.ps1

```diff
@@ -0,0 +1,192 @@
+using module PSKoans
+[Koan(Position = 104)]
+param()
+<#
+    Object-Orientated Programming (OOP)
```

Comment:
I've had a good read through this file, and I'm not sure it's something we need to cover here directly. This is pretty high-level lecture-type stuff, which is probably a bit more likely to confuse and deter newbies rather than encourage them, from what I've seen.

The concepts here are valuable, but someone just starting out needs to know a lot more of the "how" than the "why" it works the way it does. These topics are generally more easily covered in a blog post / classroom lesson format, and don't really bear a direct relation to _using_ PowerShell...

I'm not sure there's a good way to frame something like this for the koans... while I agree it's something everyone should have some awareness of, I'm not sure it's something everyone needs in great detail like we have here.

If you want, you can boil it town to basics and make a much shortened advice file out of it, maybe with links to some more thorough reading on the topic? 🙂

Reply:
I'd say it's fundamental to working with PowerShell. I've taught 100s of individuals in my time across multiple companies, and it's only when an individual understands these principles that they can start utilising PowerShell to its full potential. The "how" element being loosely covered in "AboutGetMember" and "AboutCmdletVerbs". Of course we're not even scratching the surface of OOP, but it's enough for someone just using PowerShell. 

I can make an advise file sure, but I can confidently state that in my own professional opinion the Koan should stand as it is. In my teaching experience boiling this down to relatable topics (OK the Coventry example is very much a UK reference but still) helps demystify the whole concept.

Of course you're moderating the PR, so I'll convert it to an advise file if required to get the PR merged.

Reply:
Well, like I said, the concepts are valuable; I don't think you're wrong at all, just not sure this format/design really fits the koans well. I kind of do want to keep it, but it doesn't feel finished to me... but equally, I'm not sure I can suggest a better way to do it at the moment? 😕 

If you don't mind, I'd like you to make an advice file out of it so we have some semblance of the concept "covered" in the module for the time being. But I'd also like you to leave a copy of the original you have here in the `Drafts` folder in the repo so you, me, or others can pick it up to revisit it later, if that's cool with you? 😊 

You're right that this could be helpful for a lot of users, but I don't think this is quite in the right place just yet.

Reply:
I've transferred it across to an advise file as best as I can, and move the original Koan to the "Drafts" folder.

Also:
- Added an external link to OOP principles in the advise file 
- Moved AboutOOPAnswers to drafts folder
- Deleted AnswerFunctions folder

Let me know if it's alright!